pennylane/templates/layers.py

```diff
@@ -0,0 +1,418 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Layers
+======
+
+**Module name:** :mod:`pennylane.templates.layers`
+
+.. currentmodule:: pennylane.templates.layers
+
+This module contains templates for trainable `layers`. A layer is a fixed sequence of quantum gates,
+where some gates have trainable gate parameters (like rotation angles or squeezing amplitudes). This
+makes the layer `learnable` within the limits of the architecture. A variational circuit typically repeats layers.
+
+Most templates in this module have a ``Layer`` version that implements a single layer, as well as a ``Layers``
+version which calls the single layer multiple times, possibly using different hyperparameters for the
+sequence in each call.
+
+
+Qubit architectures
+-------------------
+
+Strongly entangling circuit
+***************************
+
+.. autosummary::
+
+    StronglyEntanglingLayers
+    StronglyEntanglingLayer
+
+Random circuit
+**************
+
+.. autosummary::
+
+    RandomLayers
+    RandomLayer
+
+Continuous-variable architectures
+---------------------------------
+
+Continuous-variable quantum neural network
+******************************************
+
+.. autosummary::
+
+    CVNeuralNetLayers
+    CVNeuralNetLayer
+
+Interferometer
+**************
+
+.. autosummary::
+
+    Interferometer
+
+Code details
+^^^^^^^^^^^^
+"""
+#pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from collections.abc import Sequence
+
+from pennylane.ops import CNOT, RX, RY, RZ, Rot, Squeezing, Displacement, Kerr, Beamsplitter, Rotation
+from pennylane.qnode import QuantumFunctionError
+from pennylane.variable import Variable
+import numpy as np
+
+
+def StronglyEntanglingLayers(weights, ranges=None, imprimitive=CNOT, wires=None):
```

Comment:
the example code in templates.rst has an additional parameter 'periodic=True' which is missing here

Reply:
Fixed